Sources/Shared/ImageManager.swift

```diff
@@ -26,15 +27,16 @@ public class ImageManager {
       return
     }
 
-    Configuration.imageCache.async.object(forKey: url.absoluteString) { [weak self] (object: Image?) in
+    Configuration.imageStorage?.async.object(ofType: ImageWrapper.self,
+                                            forKey: url.absoluteString) { [weak self] result in
       guard let `self` = self else {
         return
       }
 
       // Return image from cache if it exists.
-      if let image = object {
+      if case .value(let wrapper) = result {
```

Comment:
I think we it's good to call completion in the case of error as well.

Reply:
@vadymmarkov: The signature is `public typealias Completion = (Image?) -> Void` so I think we should fix that in the next PR